tests/maintenance_test.go

```diff
@@ -0,0 +1,128 @@
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("deis maintenance", func() {
+
+	Context("with an existing user", func() {
+
+		var user model.User
+
+		BeforeEach(func() {
+			user = auth.Register()
+		})
+
+		AfterEach(func() {
+			auth.Cancel(user)
+		})
+
+		Context("who owns an existing app that has already been deployed", func() {
+
+			var app model.App
+
+			BeforeEach(func() {
+				app = apps.Create(user, "--no-remote")
+				builds.Create(user, app)
+			})
+
+			AfterEach(func() {
+				apps.Destroy(user, app)
+			})
+
+			Specify("can list that app's maintenance info", func() {
+				sess, err := cmd.Start("deis maintenance:info --app=%s", &user, app.Name)
+				Eventually(sess).Should(Say("Maintenance mode is off.\n"))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+			})
+
+			Specify("can view app when maintenance mode is off", func() {
+				// curl the app's root URL and print just the HTTP response code
+				cmdRetryTimeout := 60
+				curlCmd := model.Cmd{CommandLineString: fmt.Sprintf(`curl -sL -w "%%{http_code}\\n" "%s" -o /dev/null`, app.URL)}
+				Eventually(cmd.Retry(curlCmd, strconv.Itoa(200), cmdRetryTimeout)).Should(BeTrue())
+			})
+
+			Specify("can enable/disable maintenance", func() {
+				sess, err := cmd.Start("deis maintenance:on --app=%s", &user, app.Name)
+				Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("done"))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+
+				sess, err = cmd.Start("deis maintenance:info --app=%s", &user, app.Name)
+				Eventually(sess).Should(Say("Maintenance mode is on.\n"))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+
+				// curl the app's root URL and print just the HTTP response code
+				cmdRetryTimeout := 60
+				curlCmd := model.Cmd{CommandLineString: fmt.Sprintf(`curl -sL -w "%%{http_code}\\n" "%s" -o /dev/null`, app.URL)}
+				Eventually(cmd.Retry(curlCmd, strconv.Itoa(503), cmdRetryTimeout)).Should(BeTrue())
+
+				sess, err = cmd.Start("deis maintenance:off --app=%s", &user, app.Name)
+				Eventually(sess, settings.MaxEventuallyTimeout).Should(Say("done"))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+
+				sess, err = cmd.Start("deis maintenance:info --app=%s", &user, app.Name)
+				Eventually(sess).Should(Say("Maintenance mode is off.\n"))
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess).Should(Exit(0))
+
+				cmdRetryTimeout = 60
+				curlCmd = model.Cmd{CommandLineString: fmt.Sprintf(`curl -sL -w "%%{http_code}\\n" "%s" -o /dev/null`, app.URL)}
+				Eventually(cmd.Retry(curlCmd, strconv.Itoa(200), cmdRetryTimeout)).Should(BeTrue())
+			})
+		})
+	})
+
+	DescribeTable("any user can get command-line help for maintenance", func(command string, expected string) {
```

Comment:
every other command currently has e2e tests for this, though I agree we can just leave this to cli/sdk unit tests instead of e2e to make e2e more stable/faster.